File: BACK/backend/Proyecto-Gestion-Ambiental/databasefirst/Models/Iniciativa.cs
using System;
using System.Collections.Generic;

namespace databasefirst.Models;

public partial class Iniciativa
{
    public int id_iniciativa { get; set; }

    public string nombre { get; set; } = null!;

    public string? descripcion { get; set; }

    public DateOnly? fecha_inicio { get; set; }

    public DateOnly? fecha_fin { get; set; }

    public string? estado { get; set; }

    public int? id_categoria { get; set; }

    public virtual ICollection<Actividad> Actividads { get; set; } = new List<Actividad>();

    public virtual Categorium? id_categoriaNavigation { get; set; }
}

File: FRONT/frontend/proyecto-Gestion-Ambiental/src/app/models/Iniciativa.ts
//  <!--Nombre del autor: Sebastian Mendoza-->
export interface Iniciativa {
  id_iniciativa: number;
  nombre: string;
  descripcion: string;        
  fecha_inicio: string;
  fecha_fin: string;                  
  estado: 'Planificada' | 'En Progreso' | 'Finalizada' | 'Activo'; 
  id_categoria: number; 
  
  id_categoriaNavigation?: {
    nombre: string;
    descripcion: string;
    estado: string;
  };
}